Mathlib/NumberTheory/DirichletCharacter/Basic.lean

```diff
@@ -91,9 +91,9 @@ lemma changeLevel_trans {m d : ℕ} (hm : n ∣ m) (hd : m ∣ d) :
 
 lemma changeLevel_eq_cast_of_dvd {m : ℕ} (hm : n ∣ m) (a : Units (ZMod m)) :
     (changeLevel hm χ) a = χ (ZMod.cast (a : ZMod m)) := by
-  set_option tactic.skipAssignedInstances false in
-  simpa [changeLevel_def, Function.comp_apply, MonoidHom.coe_comp] using
-      toUnitHom_eq_char' _ <| ZMod.isUnit_cast_of_dvd hm a
+  simp only [changeLevel_def, toUnitHom_eq, ofUnitHom_eq, equivToUnitHom_symm_coe,
+    MonoidHom.coe_comp, Function.comp_apply, coe_equivToUnitHom]
+  rfl
```

Comment:
Missing lemma?

Reply:
Yes, there is a missing lemma about `ZMod.unitsMap`:
One can go from `Units (ZMod 10)` to `ZMod 5` either via `ZMod 10` or via `Units (ZMod 5)`. I've added it like this:
```
@[simp]
def val_unitsMap {m n : Nat} (h : n ∣ m) (a : Units (ZMod m)) :
    (ZMod.unitsMap h a).val = (a.val).cast := rfl
```
This works to fix this `simp`, but it is non-confluent in combination with `[MulChar.one_apply_coe](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/MulChar/Basic.html#MulChar.one_apply_coe)`. This caused the `simp` in `DirichletCharacter.changeLevel` to break down.

But I fixed that `simp` by changing the simp normal form of `MulChar.EquivToUnitHom` to be `MulChar.mulEquivToUnitHom`, so that it simplifies away the 1 more easily.